test/api/test_models_api.py

```diff
@@ -62,3 +64,166 @@ def test_dafni_get_request_called_correctly(self, mock_get):
         mock_get.assert_called_once_with(
             MODELS_API_URL + "/models/version_1/definition/", JWT
         )
+
+
+@patch("dafni_cli.api.models_api.dafni_put_request")
+@patch(
+        "builtins.open", new_callable=mock_open, read_data="definition file"
+    )
+class TestValidateModelDefinition:
+    """Test class to test the validate_model_definition functionality"""
+
+    def test_valid_model_definition_file_processed_correctly(
+            self,
+            open_mock,
+            mock_put,
+            request_response_fixture
+    ):
+        # SETUP
+        request_response_fixture.json.return_value = {"valid": True}
+        mock_put.return_value = request_response_fixture
+        mock_file = Path("definition_file")
+        jwt = "JWT"
+
+        # CALL
+        response, error_message = models_api.validate_model_definition(jwt, mock_file)
+
+        # ASSERT
+        open_mock.assert_called_once_with(
+            mock_file, "rb"
+        )
+        mock_put.assert_called_once_with(
+            MODELS_API_URL + "/models/definition/validate/", jwt, open(mock_file, "rb"), VALIDATE_MODEL_CT
+        )
+        assert response
+        assert error_message == ""
+
+    def test_invalid_model_definition_file_processed_correctly(
+            self,
+            open_mock,
+            mock_put,
+            request_response_fixture
+    ):
+        # SETUP
+        request_response_fixture.json.return_value = {"valid": False, "errors": ["error message"]}
+        mock_put.return_value = request_response_fixture
+        mock_file = Path("definition_file")
+        jwt = "JWT"
+
+        # CALL
+        response, errors = models_api.validate_model_definition(jwt, mock_file)
+
+        # ASSERT
+        open_mock.assert_called_once_with(
+            mock_file, "rb"
+        )
+        mock_put.assert_called_once_with(
+            MODELS_API_URL + "/models/definition/validate/", jwt, open(mock_file, "rb"), VALIDATE_MODEL_CT
+        )
+        assert not response
+        assert errors == "error message"
+
+
+@patch("dafni_cli.api.models_api.dafni_post_request")
+class TestGetModelUploadUrls:
+    """Test class to test the get_model_upload_urls functionality"""
+
+    def test_post_request_called_with_correct_arguments(self, mock_post):
+        # SETUP
+        jwt = "JWT"
+        url = MODELS_API_URL + "/models/upload/"
+        data = {"image": True, "definition": True}
+
+        # CALL
+        _, _ = models_api.get_model_upload_urls(jwt)
```

Comment:
this isnt necessary, you can just do `models_api.get_model_upload_urls(jwt)`

Reply:
Changed